gazebo_plugins/src/gazebo_ros_wheel_slip.cpp

```diff
@@ -50,21 +63,22 @@ void GazeboRosWheelSlip::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
   // Initialize the WheelSlipPlugin first so its values are preferred unless the ros
   // parameters are overridden by a launch file.
   WheelSlipPlugin::Load(_model, _sdf);
-  double slip_lateral_default = 0.0;
-  double slip_longitudinal_default = 0.0;
 
+  // Read slip values from the sdf file and set them as default
   if (_sdf->HasElement("wheel")) {
-    auto wheel_element = _sdf->GetElement("wheel");
-    while (wheel_element) {
+    for (auto wheel_element = _sdf->GetElement("wheel"); wheel_element;
+      wheel_element = wheel_element->GetNextElement("wheel"))
+    {
+      auto wheel_name = wheel_element->Get<std::string>("link_name");
       double slip_lateral = wheel_element->Get<double>("slip_compliance_lateral");
-      if (slip_lateral >= 0.) {
-        slip_lateral_default = slip_lateral;
-      }
+      double default_slip_lateral = std::max(slip_lateral, 0.);
```

Comment:
Maybe we should log a warning if a user provides a negative value; noting that we're clamping to zero.